### PR TITLE
[Implement][iMarari]Automation School task 2 (OOP)

### PR DESCRIPTION
**Pull request by iMarari**
1 iteration of 2 task (oop)

### CODE REVIEW

File: iuriMorari/src/main/java/AbstractOrder.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        return lumpAmount;
+    }
+
+    public double sum(Item itemToFilter) {
+        double lumpAmount = 0;
+        for (Item item : this.items) {
+            if (item.getTitle().equalsIgnoreCase(itemToFilter.getTitle())) {
+                lumpAmount += itemToFilter.getPrice();
+            }
+
```
Comment (asecu) [inline iuriMorari/src/main/java/AbstractOrder.java]: remove empty lines please

File: iuriMorari/src/main/java/AbstractUser.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public String toString() {
+        return "AbstractUser{" +
+                "id=" + id +
+                ", username='" + username + '\'' +
+                ", addresses=" + addresses +
+                '}';
+    }
+
+    // creating methods to filter list of users whose city is Chisinau
```
Comment (asecu) [inline iuriMorari/src/main/java/AbstractUser.java]: adapt your comment please

File: iuriMorari/src/main/java/AdminUser.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    public AdminUser(String username, Map<String, Address> addresses) {
+        super(username, addresses);
+    }
+
+    public AdminUser(String username) {
+        super(username);
+    }
+
+    public void approveOrder(AbstractOrder userOrder) {
```
Comment (asecu) [inline iuriMorari/src/main/java/AdminUser.java]: the variable name is not suggestive, let's use appropriate abstractOrder name

File: iuriMorari/src/main/java/AdminUser.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        System.out.println("Order " + userOrder.anyID() + " is approved");
```
Comment (asecu) [inline iuriMorari/src/main/java/AdminUser.java]: in such case you will print out an unknown ID, make shure that you are going to print out the manipulated order ID

File: iuriMorari/src/main/java/AdminUser.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    public void rejectOrder(AbstractOrder userOrder) {
+        System.out.println("Order " + userOrder.anyID() + " is rejected");
```
Comment (asecu) [inline iuriMorari/src/main/java/AdminUser.java]: the same as above

File: iuriMorari/src/main/java/Company.java [hunk 6]
```diff
@@ -0,0 +1,3 @@
+public class Company {
+
```
Comment (asecu) [inline iuriMorari/src/main/java/Company.java:4]: add at leas one attribute here)

File: iuriMorari/src/main/java/Item.java [hunk 7]
```diff
@@ -24,4 +24,20 @@ public String toString() {
                 ", price=" + price +
                 '}';
     }
+
```
Comment (asecu) [inline iuriMorari/src/main/java/Item.java:32]: The price can not be negative, please handle this by using if()

File: iuriMorari/src/main/java/UserOrder.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import java.util.List;
+
+public class UserOrder extends AbstractOrder {
+    public User user;
```
Comment (asecu) [inline iuriMorari/src/main/java/UserOrder.java]: can we go for private user?

File: iuriMorari/src/main/java/UserOrder.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+    public UserOrder(List<Item> items, User user) {
+        super(items);
+        this.user = user;
+    }
+}
+
+
```
Comment (asecu) [inline iuriMorari/src/main/java/UserOrder.java:20]: where is the getter and setter for our user?

**Review by asecu — CHANGES_REQUESTED** [review]

File: iuriMorari/src/main/java/AbstractUser.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        this.username = username;
+        this.addresses = addresses;
+    }
+
+    public AbstractUser(String username) {
+        this.id = anyID();
+        this.username = username;
+    }
+
+    private String anyID() {
```
Comment (solonarieugeniu) [inline iuriMorari/src/main/java/AbstractUser.java]: rename please

**Review by solonarieugeniu — COMMENTED** [review]
please use the following format for your PRs, [username] Java task, e.g. [esolonari] Exception handling

File: iuriMorari/src/main/java/AdminUser.java (hunk 4 above)
Comment (iMarari) [reply iuriMorari/src/main/java/AdminUser.java]: not sure if understood correctly, but done

File: iuriMorari/src/main/java/AdminUser.java (hunk 5 above)
Comment (iMarari) [reply iuriMorari/src/main/java/AdminUser.java]: same

File: iuriMorari/src/main/java/AbstractOrder.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+    private final LocalDateTime date;
+    private List<Item> items;
+
+    public AbstractOrder(List<Item> items) {
+        this.id = UUID.randomUUID().toString();
+        this.date = LocalDateTime.now();
+        this.items = items;
+    }
+
+    public String generatedID() {
```
Comment (asecu) [inline iuriMorari/src/main/java/AbstractOrder.java]: we are not using this method, let's remove it

File: iuriMorari/src/main/java/AbstractOrder.java [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    public String toString() {
+        return "AbstractOrder{" +
+                "id='" + id + '\'' +
+                ", date=" + date +
+                ", items=" + items +
+                '}';
+
+    }
+
+    //creating function to filter list of orders by type
```
Comment (asecu) [inline iuriMorari/src/main/java/AbstractOrder.java]: java is using methods, adapt your comment please

File: iuriMorari/src/main/java/AbstractUser.java [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    private Map<String, Address> addresses;
+
+    public AbstractUser(String username, Map<String, Address> addresses) {
+        this.username = username;
+        this.addresses = addresses;
+    }
+
+    public AbstractUser(String username) {
+        this.username = username;
+    }
```
Comment (asecu) [inline iuriMorari/src/main/java/AbstractUser.java]: Still we use both methods?

File: iuriMorari/src/main/java/Item.java [hunk 14]
```diff
@@ -13,6 +13,7 @@ public Item(String title, Type type, Double price) {
     }
 
     public Item(String boots, Type boots1, int i) {
```
Comment (asecu) [inline iuriMorari/src/main/java/Item.java]: This constructor should be refactored. The declared variables within the method should be used.

File: iuriMorari/src/main/java/AbstractOrder.java (hunk 11 above)
Comment (iMarari) [reply iuriMorari/src/main/java/AbstractOrder.java]: It is used in "AdminUser" Class on printing approveOrder and rejectOrder.

image

File: iuriMorari/src/main/java/AdminUser.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
+import java.util.Map;
+
+public class AdminUser extends AbstractUser {
+
+    public AdminUser(String username, Map<String, Address> addresses) {
+        super(username, addresses);
+    }
+
+    public void approveOrder(AbstractOrder abstractOrder) {
+        System.out.println("Order " + abstractOrder.generatedID() + " is approved");
```
Comment (solonarieugeniu) [inline iuriMorari/src/main/java/AdminUser.java]: id should be of the order to be approved/rejected, why do you generate it here?

**Review by solonarieugeniu — APPROVED** [review]

File: iuriMorari/src/main/java/AbstractUser.java (hunk 13 above)
Comment (iMarari) [reply iuriMorari/src/main/java/AbstractUser.java]: No, deleted.

File: iuriMorari/src/main/java/AdminUser.java (hunk 15 above)
Comment (iMarari) [reply iuriMorari/src/main/java/AdminUser.java]: As I understood, now its "getId()"

File: iuriMorari/src/main/java/AbstractOrder.java (hunk 11 above)
Comment (iMarari) [reply iuriMorari/src/main/java/AbstractOrder.java]: Renamed on getID

**Review by asecu — APPROVED** [review]

**Review by gaenache — APPROVED** [review]

**Review by dvmunteanu — APPROVED** [review]

**Review by constantin73 — APPROVED** [review]

**Pull request merged**